actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -46,8 +47,160 @@ class TagBuilder # :nodoc:
         include CaptureHelper
         include OutputSafetyHelper
 
-        HTML_VOID_ELEMENTS = %i(area base br col embed hr img input keygen link meta param source track wbr).to_set
-        SVG_SELF_CLOSING_ELEMENTS = %i(animate animateMotion animateTransform circle ellipse line path polygon polyline rect set stop use view).to_set
+        def self.define_element(name, method_name: name.to_s.underscore)
+          class_eval(<<~RUBY, __FILE__, __LINE__ + 1) unless instance_methods.include?(method_name.to_sym)
+            def #{method_name}(*arguments, **options, &block)
+              tag_string(#{name.inspect}, *arguments, **options, &block)
+            end
+          RUBY
+        end
+
+        def self.define_void_element(name, method_name: name.to_s.underscore, self_closing: false)
+          class_eval <<~RUBY, __FILE__, __LINE__ + 1
+            def #{method_name}(content = nil, escape: true, **options, &block)
+              if content.nil? && block.nil?
+                void_tag_string(#{name.inspect}, options, escape, #{self_closing})
+              else
+                tag_string(#{name.inspect}, content, escape: escape, **options, &block)
+              end
+            end
+          RUBY
+        end
+
+        def self.define_self_closing_element(name, **options)
+          define_void_element(name, self_closing: true, **options)
+        end
+
+        define_void_element :area
+        define_void_element :base
+        define_void_element :br
+        define_void_element :col
+        define_void_element :embed
+        define_void_element :hr
+        define_void_element :img
+        define_void_element :input
+        define_void_element :keygen
+        define_void_element :link
+        define_void_element :meta
+        define_void_element :source
+        define_void_element :track
+        define_void_element :wbr
+
+        define_self_closing_element :animate
+        define_self_closing_element :animateMotion
+        define_self_closing_element :animateTransform
+        define_self_closing_element :circle
+        define_self_closing_element :ellipse
+        define_self_closing_element :line
+        define_self_closing_element :path
+        define_self_closing_element :polygon
+        define_self_closing_element :polyline
+        define_self_closing_element :rect
+        define_self_closing_element :set
+        define_self_closing_element :stop
+        define_self_closing_element :use
+        define_self_closing_element :view
+
+        define_element :a
```

Comment:
@rafaelfranca which do you think would make the most sense:

1. Split the definitions into void, self-closing, then the rest
2. Alphabetize _all_ the definitions in a single group, with the void, self-closing, and rest inter-mingled

If 2. feels more preferable, do you think there should be a single `define_element` class method that takes option arguments (e.g. `define_element :div`, `define_element :area, void: true`, `define_element :circle, self_closing: true`)?

I think my preference is 1. (the current groupings).